dbt/adapters/spark_cde/cdeapisession.py

```diff
@@ -388,6 +376,61 @@ def getJobOutput(self, job):
                 print(traceback.format_exc())
 
         return schema, rows
+    
+    def parseEventResult(self, res_lines):
+        events = []
+        
+        for event_line in res_lines:
+            # logger.debug("parseEventResult event line {}".format(event_line))
+            if len(event_line.strip()):
+                json_rec = json.loads(event_line)
+                if ('Timestamp' in json_rec):
+                    events.append(json_rec)
+                
+        return events
+
+    def getJobOutput(self, job_name, job, log_type="stdout"):  # log_type can be "stdout", "stderr", "event"
+        time_slept_while_polling_logresults = 0
+        res_lines = []
+        
+        no_of_retries = 0
+
+        while len(res_lines) <= MIN_LINES_TO_PARSE:  # ensure that enough lines are present to start parsing
+            req_url = self.base_api_url + "job-runs" + "/" + repr(job["id"]) + "/logs?type=driver%2F" + log_type + "&follow=true"
+            res = requests.get(req_url, headers=self.api_header)
+
+            # print("getJobOutput - url", req_url)
+            # print("getJobOutput - job", job)
+            # print("getJobOutput - res", res)
+            # print("getJobOutput - res.text", res.text)
+
+            # parse the o/p for data
+            res_lines = list(map(lambda x: x.strip(), res.text.split("\n")))
+
+            # print("getJobOutput - lines", res_lines, len(res_lines))
+
+            n_lines = len(res_lines)
+            if (n_lines > MIN_LINES_TO_PARSE): break  # we have some o/p to process, break out - what happens for CREATE stm?
```

Comment:
This logic is still flaky. We have few lines to parse doesn't mean that we have all the required output populated to logs to parse. Needs improvement.

Reply:
Unfortunately there doesn't seem to be a way to know that all the log is available.